libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginBaseDiscoveryFragment.java

```
@@ -38,6 +38,10 @@ void initiateDiscovery() {
     @SuppressWarnings("unused")
     @Subscribe(threadMode = ThreadMode.MAIN)
     public void onDiscoverySucceeded(OnDiscoveryResponse event) {
+        if (mLoginBaseDiscoveryListener == null) {
```

This change fixes the crash, but what does the user see when the listener is null? It seems like it fails silently, so the user doesn't know what happened.

---

Good point! I noticed this is what is happening in a lot of the login error scenarios. If the [`mRequestedSiteAddress` is `null`](https://github.com/woocommerce/woocommerce-android/blob/d8f27b82031a4036e524c1c5cb80f5b6baa241a9/libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginBaseDiscoveryFragment.java#L48) or if the [fragment is not yet added to its activity](https://github.com/woocommerce/woocommerce-android/blob/d8f27b82031a4036e524c1c5cb80f5b6baa241a9/libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginBaseDiscoveryFragment.java#L52), it is assumed that the user bailed and they will need to initiate the request again. It doesn't seem ideal but since this behaviour is being used in a lot of places in Wp-android discovery process and only a couple of user faced this issue in over a month in Woo but (and none in WP), so I _thought_ the chances of this happening is very low and following the same behaviour for the crash.

---

> so I _thought_ the chances of this happening is very low and following the same behaviour for the crash

Fair enough! I think we're good to :shipit: